src/Nethermind.Int256/UInt256.cs

```diff
@@ -1882,9 +1882,11 @@ public void Convert(out BigInteger big)
 
         public static UInt256 Parse(in ReadOnlySpan<char> value, NumberStyles numberStyles) => !TryParse(value, numberStyles, CultureInfo.InvariantCulture, out UInt256 c) ? throw new FormatException() : c;
 
-        public static bool TryParse(string value, out UInt256 result) => TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out result);
+        public static bool TryParse(string value, out UInt256 result) => TryParse(value.AsSpan(), out result);
 
-        public static bool TryParse(ReadOnlySpan<char> value, out UInt256 result) => TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out result);
+        public static bool TryParse(ReadOnlySpan<char> value, out UInt256 result) => value.StartsWith("0x", StringComparison.OrdinalIgnoreCase)
+            ? TryParse(value.Slice(2), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out result)
+            : TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out result);
```

Comment:
I am wondering about the ones that are not '0x' prefixed, can we support them too?

Reply:
Not really - only in some cases which may lead to much confusion (e. g. we track `A,B,C,D,E,F`, then what is `"22222"` - a hex or a decimal...if we treat it as hex as a decimal it will become `2236962` and this is not good).